pkg/controller/garbagecollector/garbagecollector.go

```diff
@@ -451,15 +451,27 @@ func monitorFor(p *Propagator, clientPool dynamic.ClientPool, resource unversion
 				// namespaces if it's namespace scoped, so leave
 				// APIResource.Namespaced as false is all right.
 				apiResource := unversioned.APIResource{Name: resource.Resource}
-				return client.Resource(&apiResource, api.NamespaceAll).List(&options)
+				// The parameter codec used by the dynamic client cannot encode
```

Comment:
Actually `opts.GetObjectKind().GroupVersionKind()` might be empty, because we have the tradition of resetting the gvk in memory. Also when creating via struct literal, people usually ignore setting the gvk: https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/namespace/namespace_controller_utils.go#L156.